### PR TITLE
Filter register parathread calls starlight & flashbox

### PR DESCRIPTION
**Pull request by girazoki**
This PR filters out **_register_parathread_** calls in flashbox & starlight and additionally it filters **_register_** & **_deregister_** calls from the relay registrar pallet in starlight.

**_Flashbox motivation_**: registering parathreads in flashbox does not make much sense as XCM is not enabled and therefore parathreads are not manageable.

**_Starlight motivation_**: parathreads are not yet supported in starlight. Additionally, registrations  & deregistrations should **_only_** happen through the `containerRegistrar` and not through the relay registrar

### CODE REVIEW

**Review by tmpolaczyk — APPROVED** [review]

File: solo-chains/runtime/starlight/src/lib.rs [hunk 1]
```diff
@@ -202,14 +202,41 @@ pub fn native_version() -> NativeVersion {
...
+/// Everything is handled by the containerRegistrar
+pub struct IsRelayRegister;
+impl Contains<RuntimeCall> for IsRelayRegister {
+    fn contains(c: &RuntimeCall) -> bool {
+        matches!(
+            c,
+            RuntimeCall::Registrar(paras_registrar::Call::register { .. })
+        ) || matches!(
+            c,
+            RuntimeCall::Registrar(paras_registrar::Call::deregister { .. })
```
[the remark this answers is not in the document]
Comment (tmpolaczyk) [reply solo-chains/runtime/starlight/src/lib.rs:215]: And what about filtering all the extrinsics of the pallet? Are there any useful ones we don't want to filter?

[the remark this answers is not in the document]
Comment (Agusrodri) [reply solo-chains/runtime/starlight/src/lib.rs:215]: These are all the extrinsics of the pallet. Probably we can filter them all yes.
```rust
pub trait WeightInfo {
	fn reserve() -> Weight;
	fn register() -> Weight;
	fn force_register() -> Weight;
	fn deregister() -> Weight;
	fn swap() -> Weight;
	fn schedule_code_upgrade(b: u32) -> Weight;
	fn set_current_head(b: u32) -> Weight;
}
```

[the remark this answers is not in the document]
Comment (girazoki) [reply solo-chains/runtime/starlight/src/lib.rs:215]: no, `reserve` we need it, that is how people reserve a para-id. swap we might. The rest are management things that can only be done with sudo, so even if you filter them, they are still callable by sudo

[the remark this answers is not in the document]
Comment (Agusrodri) [reply solo-chains/runtime/starlight/src/lib.rs:214]: Ah yes, I thought that at the end we were not going to make use of relay `reserve`, and just pass an arbitrary paraId. In that case the current filter is fine.

[the remark this answers is not in the document]
Comment (Agusrodri) [reply solo-chains/runtime/starlight/src/lib.rs:214]: disregard

**Review by Agusrodri — APPROVED** [review]

**Pull request merged**